### PR TITLE
implement `select`, `gernerate plist`

### LINKED ISSUE #5: XCode Target + plist gen
## 1. `PRODUCT_MODULE_NAME`

When new project created. It contain a special key `$(PRODUCT_MODULE_NAME)` in plist.
replace it with `$(PRODUCT_NAME)`

```xml
<key>UISceneDelegateClassName</key>
<string>$(PRODUCT_MODULE_NAME).SceneDelegate</string>
```

## 2. `GENERATE_INFOPLIST_FILE`

When `GENERATE_INFOPLIST_FILE: YES`. All of XCode Setting which contains prefix `INFOPLIST_KEY_` must perform auto generation.

## XCode Setting

```yaml
INFOPLIST_KEY_UISupportedInterfaceOrientations_iPad:
  UIInterfaceOrientationPortrait UIInterfaceOrientationPortraitUpsideDown UIInterfaceOrientationLandscapeLeft UIInterfaceOrientationLandscapeRight
```

## plist auto gen

```xml
<key>UISupportedInterfaceOrientations~ipad</key>
<array>
	<string>UIInterfaceOrientationPortrait</string>
	<string>UIInterfaceOrientationLandscapeLeft</string>
	<string>UIInterfaceOrientationLandscapeRight</string>
</array>
```

## 3. `Default Value??`

When plist didn't define `CFBundleName`.

```yaml
CFBundleName $(PRODUCT_NAME)
CFBundleIdentifier $(PRODUCT_BUNDLE_IDENTIFIER)
CFBundleVersion $(CURRENT_PROJECT_VERSION)
CFBundleExecutable $(EXECUTABLE_NAME)
CFBundlePackageType $(PRODUCT_BUNDLE_PACKAGE_TYPE)
CFBundleDevelopmentRegion $(DEVELOPMENT_LANGUAGE)
CFBundleShortVersionString $(MARKETING_VERSION)
CFBundleInfoDictionaryVersion 6.0
LSRequiresIPhoneOS YES
```

### LINKED ISSUE #20: XCode Target + Config(deug/release/...) use `select`

### PR DESCRIPTION
**Pull request by yume190**
add config function `select` & `preffer`
add rule `plist_fragment`
add plist implementation

### CODE REVIEW

File: Sources/BazelizeKit/Bazel/WORKSPACE+Builder.swift [hunk 1]
```diff
@@ -119,6 +119,19 @@ extension Workspace {
...
+        mutating
+        public func rulesPlistFragment() {
+            _code = """
+            load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
+            git_repository(
+                name = "Plist",
+                commit = "259ca0a5d77833728c18fa6365285559ce8cc0bf",
+                remote = "https://github.com/imWildCat/MinimalBazelFrameworkDemo",
+            )
+
```
Comment (MikeChongCan) [inline Sources/BazelizeKit/Bazel/WORKSPACE+Builder.swift:131]: is this extra empty line needed?

File: Sources/BazelizeKit/Codegen/Codegen+Application.swift [hunk 2]
```diff
@@ -6,110 +6,137 @@
...
+            "resources" => {
+                resources
+            }
             StarlarkProperty.Visibility.public
         }
     }
 
     /// macos_application(name, additional_contents, additional_linker_inputs, app_icons, bundle_extension, bundle_id, bundle_name, codesign_inputs, codesignopts, deps, entitlements, entitlements_validation, executable_name, exported_symbols_lists, extensions, include_symbols_in_bundle, infoplists, ipa_post_processor, linkopts, minimum_deployment_os_version, minimum_os_version, platform_type, provisioning_profile, resources, stamp, strings, version, xpc_services)
-    func buildMac(_ builder: inout Build.Builder) {
+    func buildMac(_ builder: inout Build.Builder, _: Kit) {
```
Comment (MikeChongCan) [inline Sources/BazelizeKit/Codegen/Codegen+Application.swift:75]: `inout` used here is not perfect for Swift concurrency. Maybe in the future we need to make it more functional (replace `inout` with others)?

File: Sources/BazelizeKit/Codegen/Codegen+Config.swift [hunk 3]
```diff
@@ -13,11 +13,11 @@ import XCode
 
 /// https://bazel.build/docs/configurable-attributes
 extension Project {
-    // MARK: Public
+    // MARK: Internal
 
     /// build:debug --//:mode=debug
     /// bazel build --config=debug [PACKAGE:RULE]
-    public func generateBazelRC(_ kit: Kit) throws {
+    internal func generateBazelRC(_ kit: Kit) throws {
```
Comment (MikeChongCan) [inline Sources/BazelizeKit/Codegen/Codegen+Config.swift:20]: This `internal` is used by default. Is it added by the linter?

**Review by MikeChongCan — APPROVED** [review]

**Comment by MikeChongCan** [comment]
Thanks for your hard work on this!

One more suggestion about logging: Since the logic is increasingly complex and we may have a lot of Github Issues in the future. Do you think we should add a logging system like https://developer.apple.com/documentation/os/logging to this project?

File: Sources/BazelizeKit/Bazel/WORKSPACE+Builder.swift (hunk 1 above)
Comment (yume190) [reply Sources/BazelizeKit/Bazel/WORKSPACE+Builder.swift:131]: To separate load rule.

File: Sources/BazelizeKit/Codegen/Codegen+Config.swift (hunk 3 above)
Comment (yume190) [reply Sources/BazelizeKit/Codegen/Codegen+Config.swift:20]: no

File: Sources/BazelizeKit/Codegen/Codegen+Application.swift (hunk 2 above)
Comment (yume190) [reply Sources/BazelizeKit/Codegen/Codegen+Application.swift:75]: Can you provide the link that `inout` is not perfect for concurrency. 

Thx.

Comment (yume190) [reply Sources/BazelizeKit/Codegen/Codegen+Application.swift:75]: In Current implementation, code generation was generated at same thread.

In this situation, is `inout` still influence `Swift concurrency`?

**Comment by yume190** [comment]
add config function `select` & `preffer`
add rule `plist_fragment`
add plist implementation

File: Sources/BazelizeKit/Codegen/Codegen+Application.swift (hunk 2 above)
Comment (MikeChongCan) [reply Sources/BazelizeKit/Codegen/Codegen+Application.swift:75]: - https://stackoverflow.com/questions/70748073/actor-isolated-property-cannot-be-passed-inout-to-async-function-call
- https://developer.apple.com/forums/thread/91694
`inout` is like a pointer here, which may block us from using concurrency features in the future.

File: Sources/BazelizeKit/Codegen/Codegen+Config.swift (hunk 3 above)
Comment (MikeChongCan) [reply Sources/BazelizeKit/Codegen/Codegen+Config.swift:20]: if it was not added by the linter, could we remove `internal`?

Comment (yume190) [reply Sources/BazelizeKit/Codegen/Codegen+Config.swift:20]: ok

File: Sources/BazelizeKit/Codegen/Codegen+Application.swift (hunk 2 above)
Comment (yume190) [reply Sources/BazelizeKit/Codegen/Codegen+Application.swift:75]: thx.

**Pull request merged**